### DISCUSSION TITLE
Webchat app on Android powered by LLama.cpp

### DISCUSSION THREAD

**Discussion by hfassold** [ggml-org/llama.cpp#3250] [Q&A]
What is the best / easiest / fastest way to get a Webchat app on Android running, which is powered by llama.cpp ?

I suppose the fastest way is via the 'server' application in combination with Node.Js 
As given in https://github.com/ggerganov/llama.cpp/tree/master/examples/server

Am I right ?

**Comment by FSSRepo** [comment]
yes, with React native and you can fetch with the `completion` endpoint.

**Comment by Green-Sky** [comment]
The server already has a "web app".

**Reply by Green-Sky** [reply]
![image](https://github.com/ggerganov/llama.cpp/assets/2938071/25850654-3d8b-428e-b012-ad4102d4df4d)

**Reply by hfassold** [reply]
Thank you ! I suppose that is the 'chat.mjs' script.

What are the  'completion.js.hpp' and 'index.js.hpp'/'index.html.hpp' files for ?
They contain only hexcode, strange.

And how do I run the server executable on a standard (non-rooted) Android phone ? Do I have to use 'termux' ?

**Reply by Green-Sky** [reply]
> What are the 'completion.js.hpp' and 'index.js.hpp'/'index.html.hpp' files for ?
They contain only hexcode, strange.

they are just the html/js sources packaged as a C array.

> And how do I run the server executable on a standard (non-rooted) Android phone ? Do I have to use 'termux' ?

yes that is what i did.

**Reply by hfassold** [reply]
Thank you. Actually, it seems that the web app is in subdirectory 'public' ('examples/server/public').
